### ISSUE TITLE
Crash when disconnecting signal

### ISSUE THREAD

**Issue by menip** [godotengine/godot#19066]
**Godot version:**
9a5950a

**Issue description:**
Clicked disconnect button on a signal.
Not terribly helpful, but maybe someone will of come across this as well.

**Steps to reproduce:**
N/A

<details>
<summary>Backtrace</summary>

```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x354b0) [0x7f0cd42824b0] (??:0)
[2] Vector<TreeItem::Cell>::size() const (godot/core/vector.h:115)
[3] TreeItem::get_metadata(int) const (godot/scene/gui/tree.cpp:304 (discriminator 2))
[4] ConnectionsDock::_make_or_edit_connection() (godot/editor/connections_dialog.cpp:439 (discriminator 2))
[5] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (godot/core/method_bind.gen.inc:56 (discriminator 4))
[6] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (godot/core/object.cpp:914 (discriminator 1))
[7] Object::emit_signal(StringName const&, Variant const**, int) (godot/core/object.cpp:1195 (discriminator 1))
[8] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/core/object.cpp:1241)
[9] ConnectDialog::ok_pressed() (godot/editor/connections_dialog.cpp:109 (discriminator 2))
[10] AcceptDialog::_ok_pressed() (godot/scene/gui/dialogs.cpp:387)
[11] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (godot/core/method_bind.gen.inc:56 (discriminator 4))
[12] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (godot/core/object.cpp:914 (discriminator 1))
[13] Object::emit_signal(StringName const&, Variant const**, int) (godot/core/object.cpp:1195 (discriminator 1))
[14] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/core/object.cpp:1241)
[15] BaseButton::_gui_input(Ref<InputEvent>) (godot/scene/gui/base_button.cpp:139 (discriminator 2))
[16] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (godot/core/method_bind.gen.inc:729 (discriminator 12))
[17] Object::call_multilevel(StringName const&, Variant const**, int) (godot/core/object.cpp:748 (discriminator 1))
[18] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/core/object.cpp:855)
[19] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (godot/scene/main/viewport.cpp:1391 (discriminator 2))
[20] Viewport::_gui_input_event(Ref<InputEvent>) (godot/scene/main/viewport.cpp:1741 (discriminator 3))
[21] Viewport::input(Ref<InputEvent> const&) (godot/scene/main/viewport.cpp:2388 (discriminator 2))
[22] Viewport::_vp_input(Ref<InputEvent> const&) (godot/scene/main/viewport.cpp:1217)
[23] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (godot/core/method_bind.gen.inc:729 (discriminator 12))
[24] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (godot/core/object.cpp:914 (discriminator 1))
[25] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/core/object.cpp:839)
[26] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (godot/scene/main/scene_tree.cpp:251)
[27] SceneTree::input_event(Ref<InputEvent> const&) (godot/scene/main/scene_tree.cpp:407 (discriminator 6))
[28] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (godot/main/input_default.cpp:392)
[29] InputDefault::parse_input_event(Ref<InputEvent> const&) (godot/main/input_default.cpp:260)
[30] OS_X11::process_xevents() (godot/platform/x11/os_x11.cpp:1892 (discriminator 3))
[31] OS_X11::run() (godot/platform/x11/os_x11.cpp:2588)
[32] godot/bin/godot.x11.tools.64(main+0xd3) [0xf59439] (godot/platform/x11/godot_x11.cpp:56)
[33] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7f0cd426d830] (??:0)
[34] godot/bin/godot.x11.tools.64(_start+0x29) [0xf59299] (??:?)
-- END OF BACKTRACE --
```

</details>

**Comment by volzhs** [comment]
can't reproduce it with 0980991e6
maybe it requires more exact steps or sample project to reproduce.

**Comment by akien-mga** [comment]
Steps to reproduce would indeed be good.

**Comment by ibrahn** [comment]
Looks like the same access-after-free I was seeing when looking into #19001.
I think #20131 fixes this.

**Comment by akien-mga** [comment]
Assuming fixed by #20131.
